### ISSUE TITLE
Cannot Render In 1.15.2 with ReplayMod 2.3.1

### ISSUE THREAD

**Issue by Kas-tle** [ReplayMod/ReplayMod#214]
I receive an immediate error when attempting to render on 1.15.2, with or without shaders.

Reproduction Steps:
1. Create a replay recording
2. Attempt to render on standard settings (1920 x 1080, 10mb/s)
3. Observe failure log.

I'm assuming this is system specific so if you need any more details (e.g. heap dumps) please let me know. 

Fabric Mods:
- replaymod-1.15.2-2.3.1
- preview_OptiFine_1.15.2_HD_U_G1_pre22
- optifabric-1.0.0-beta8
- modmenu-1.10.2+build.32
- fabric-api-0.11.1+build.312-1.15
- cmdkeybind-1.4.2

System Details:
- Fabric 1.15.2 Build 201
- MacOS 10.14.6
- Java Version 1.8.0_74, Oracle Corporation

Full error log is attached. Relevant section:
```
[00:36:42] [main/ERROR]: ---- Minecraft Crash Report ----
// Would you like a cupcake?

Time: 6/7/20 12:36 AM
Description: Rendering video

java.lang.NoSuchMethodError: java.nio.ByteBuffer.rewind()Ljava/nio/ByteBuffer;
	at com.replaymod.render.capturer.SimplePboOpenGlFrameCapturer.process(SimplePboOpenGlFrameCapturer.java:47)
	at com.replaymod.render.capturer.SimplePboOpenGlFrameCapturer.process(SimplePboOpenGlFrameCapturer.java:13)
	at com.replaymod.render.rendering.Pipeline.run(Pipeline.java:71)
	at com.replaymod.render.rendering.VideoRenderer.renderVideo(VideoRenderer.java:175)
	at com.replaymod.render.gui.GuiRenderSettings$8.run(GuiRenderSettings.java:239)
	at com.replaymod.core.ReplayMod.lambda$runLater$6(ReplayMod.java:542)
	at net.minecraft.class_1255.method_18859(class_1255.java:173)
	at net.minecraft.class_4093.method_18859(class_4093.java:23)
	at net.minecraft.class_1255.method_16075(class_1255.java:135)
	at net.minecraft.class_1255.method_5383(class_1255.java:115)
	at com.replaymod.core.ReplayMod$ReplayModExecutor.method_5383(ReplayMod.java:504)
	at com.replaymod.replay.InputReplayTimer.method_1658(InputReplayTimer.java:72)
	at com.replaymod.pathing.player.ReplayTimer.method_1658(ReplayTimer.java:33)
	at net.minecraft.class_310.method_1523(class_310.java:953)
	at net.minecraft.class_310.method_1514(class_310.java:619)
	at net.minecraft.client.main.Main.main(Main.java:204)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:192)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:140)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- System Details --
Details:
	Minecraft Version: 1.15.2
	Minecraft Version ID: 1.15.2
	Operating System: Mac OS X (x86_64) version 10.14.6
	Java Version: 1.8.0_74, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 2329760944 bytes (2221 MB) / 3657433088 bytes (3488 MB) up to 8589934592 bytes (8192 MB)
	CPUs: 8
	JVM Flags: 8 total; -Xss1M -Xmx8G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		blue_endless_jankson: jankson +
		cloth-config2: Cloth Config v2 2.8.3
		cmdkeybind: Command Macros 1.4.2
		fabric: Fabric API 0.11.1+build.312-1.15
		fabric-api-base: Fabric API Base 0.1.2+b7f9825d0c
		fabric-biomes-v1: Fabric Biomes (v1) 0.1.5+3b05f68e0c
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.4+c6a8ea890c
		fabric-climbable-api-v1: Fabric Climbable API (v1) 1.0.0+6ffe71e20c
		fabric-command-api-v1: Fabric Command API (v1) 1.0.2+2aebb7ff0c
		fabric-commands-v0: Fabric Commands (v0) 0.2.0+2e0c4ff50c
		fabric-containers-v0: Fabric Containers (v0) 0.1.3+b7f9825d0c
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.1.3+b7f9825d0c
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.2+b7f9825d0c
		fabric-dimensions-v1: fabric-dimensions-v1 0.3.0+2ad156310c
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.3.0+fac69e320c
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.1.2+003643510c
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.1.6+045efb460c
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.1.1+dfdb52d60c
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 0.1.5+e08a73050c
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.2+ec951ea40c
		fabric-models-v0: Fabric Models (v0) 0.1.0+dfdb52d60c
		fabric-networking-blockentity-v0: Fabric Networking Block Entity (v0) 0.2.3+e08a73050c
		fabric-networking-v0: Fabric Networking (v0) 0.1.7+003643510c
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.2.1+e7a4edb30c
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.5.1+3e5587510c
		fabric-particles-v1: fabric-particles-v1 0.1.2+ad2f81bd0c
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.2.6+f3d8141b0c
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.2.10+f08b61330c
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.2.23+9290e2ed0c
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 2.0.1+5a0f9a600c
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.1.3+b7f9825d0c
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.1.6+12515ed90c
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.0+534104900c
		fabric-rendering-v1: Fabric Rendering (v1) 1.0.0+2a034f0a0c
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.1.10+06c939b30c
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 0.1.3+abd915800c
		fabric-textures-v0: Fabric Textures (v0) 1.0.4+821cdba70c
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.1.0+3e5587510c
		fabricloader: Fabric Loader 0.8.7+build.201
		me_shedaniel_cloth_basic-math: basic-math 0.3.0
		minecraft: Minecraft 1.15.2
		mm: Manningham Mills 2.0
		modmenu: Mod Menu 1.10.2+build.32
		optifabric: Optifabric 1.0.0-beta8
		org_slf4j_slf4j-api: slf4j-api 1.7.30
		org_slf4j_slf4j-simple: slf4j-simple 1.7.30
		org_zeroturnaround_zt-zip: zt-zip 1.14
		replaymod: Replay Mod 1.15.2-2.3.1
	OptiFine Version: OptiFine_1.15.2_HD_U_G1_pre22
	OptiFine Build: 20200606-224145
	Render Distance Chunks: 16
	Mipmaps: 0
	Anisotropic Filtering: 1
	Antialiasing: 0
	Multitexture: false
	Shaders: null
	OpenGlVersion: 2.1 INTEL-12.10.19
	OpenGlRenderer: Intel Iris Pro OpenGL Engine
	OpenGlVendor: Intel Inc.
	CpuCount: 8
[00:37:17] [main/INFO]: Stopping!

```

[latest.log](https://github.com/ReplayMod/ReplayMod/files/4741533/latest.log)

**Comment by Johni0702** [comment]
Apparently that error can happen when the program is compiled with Java 9 but ran on Java 8.
The second part matches but we're compiling in [openjdk:8-jdk](https://hub.docker.com/layers/openjdk/library/openjdk/8-jdk/images/sha256-14dcb39ea8e39ee510cde954feccb82307278fafc75e25f7e04447ae62f26540?context=explore) which, according to its name, should be Java 8, so I'm not entirely sure what's going on.
If I run it myself using OpenJDK 1.8.0_252, I'm not getting that error, which leads me to believe that maybe this is a difference between OpenJDK and Oracle's Java (indeed I cannot find the referenced method in Oracle's JavaDocs) or between different Java 8 versions (note how you're on _74 which is fairly old compared to _252). Neither should really ever happen but it's the only thing which makes sense to me atm. Latter seems more likely since you're the first one to report such an issue.
Could you try updating your Java?

**Comment by Kas-tle** [comment]
74 is what came bundled with the launcher so that's my bad it would seem. However, implicitly telling Minecart to use the jdk I have installed (251) produces the same error. I will attempt to compile against my local java 8 install and see if the issue persists.

**Comment by Kas-tle** [comment]
Ah so my apologies. I had compiled the program myself as earlier tonight the website was inaccessible for me. It appears my terminal defaulted to openjdk 11 when compiling so this is likely the issue. I never had this problem when compiling the early versions of the 1.15 mod so I did not think to set my java version to 1.8 in the terminal before compiling. Totally my bad, sorry for taking up your time with a problem created exclusively by me. Thanks for your help.
